cmd/object-api-interface.go

```diff
@@ -67,30 +67,35 @@ type ObjectLayer interface {
 // HashReader writes what it reads from an io.Reader to an MD5 and SHA256 hash.Hash.
 // HashReader verifies that the content of the io.Reader matches the expected checksums.
 type HashReader struct {
```

Comment:
Move out `HashReader` from `object-api-interface.go` for better code wake through, may be a package if file list is limited by go

Reply:
This is unrelated to this PR, but i can do that..

Reply:
Also this change would make the PR much bigger.